### ISSUE TITLE
gnt-instance add should fail or warn if a secondary node is passed and the disk template doesn't have one

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#169]
Originally reported of Google Code with ID 113.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".
2.1.6
<b>What distribution are you using?</b>

<b>What steps will reproduce the problem?</b>
1.I want to create new instance with secondary node by command:
gnt-instance add -n x3400:hataraki -t plain --disk 0:size=4g -B memory 512 -o altbootstrap+kvm s210
Such syntax to point out the secondary node (-n primary:secondary) is described in Ganeti administrator's guide, chapter "Instance management":
gnt-instance add \
  -n TARGET_NODE:SECONDARY_NODE \
  -o OS_TYPE \
  -t DISK_TEMPLATE -s DISK_SIZE \
  INSTANCE_NAME

2. The new instance s210 has been created, but without secondary node. When I type "gnt-instance info s210", then I see output:
Nodes:
 - primary: x3400
 - secondaries:
<b>3.</b>
<b>What is the expected output? What do you see instead?</b>

What did I do wrong? Could you give me advice? If yes, hint, please, if there are any especiallities in JSON syntax for API to create instance with secondary node?

<b>Please provide any additional information below.</b>

```
Originally added on 2010-08-10 14:36:54 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

The problem here is that you're using -t plain, which doesn't support having a secondary. If you move to -t drbd (and you have drbd setup on your system) everything will work fine. I updated the title of this bug so we change ganeti to fail or warn if a secondary node is passed but is not going to be used.

Thanks,

Guido

```
Originally added on 2010-08-10 15:55:36 +0000 UTC.

Added Labels: Type-Defect Priority-Low
Added to Milestone: Release2.1

**Comment by Ganeti-Issues-Migrator** [comment]
```
Implemented in commit af5970abf5026. This is included as of Ganeti 2.1.7.
```
Originally added on 2010-09-07 11:44:15 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:08 +0000 UTC.

Changed State: Released
